src/commands.rs

```diff
@@ -475,15 +496,31 @@ where
                         }
                         _ => {
                             //TODO: something better here?
-                            return Err(e).context("error reading compile response from server");
+                            if ignore_all_server_errors() {
+                                eprintln!(
+                                    "sccache: warning: error reading compile response from server \
+                                     compiling locally instead"
+                                );
+                            }
+                            else {
+                                return Err(e).context("error reading compile response from server");
+                            }
                         }
                     }
                 }
             }
         }
         CompileResponse::UnsupportedCompiler(s) => {
             debug!("Server sent UnsupportedCompiler: {:?}", s);
-            bail!("Compiler not supported: {:?}", s);
+            if ignore_all_server_errors() {
```

Comment:
It should be a user config problem to supply an unsupported compiler. I think this is not needed.